test/infrastructure/docker/docker/types/node.go

```diff
@@ -67,23 +70,22 @@ func (n *Node) Role() (string, error) {
 
 // IP gets the docker ipv4 and ipv6 of the node.
 func (n *Node) IP(ctx context.Context) (ipv4 string, ipv6 string, err error) {
-	// retrieve the IP address of the node using docker inspect
-	cmd := exec.CommandContext(ctx, "docker", "inspect",
-		"-f", "{{range .NetworkSettings.Networks}}{{.IPAddress}},{{.GlobalIPv6Address}}{{end}}",
-		n.Name, // ... against the "node" container
-	)
-	lines, err := exec.CombinedOutputLines(cmd)
+	cli, err := dockerClient.NewClientWithOpts(dockerClient.FromEnv, dockerClient.WithAPIVersionNegotiation())
 	if err != nil {
-		return "", "", errors.Wrap(err, "failed to get container details")
+		return "", "", errors.Wrap(err, "failed to get container client")
 	}
-	if len(lines) != 1 {
-		return "", "", errors.Errorf("file should only be one line, got %d lines", len(lines))
+
+	containerInfo, err := cli.ContainerInspect(ctx, n.Name)
+	if err != nil {
+		return "", "", errors.Wrap(err, "failed to get container details")
 	}
-	ips := strings.Split(lines[0], ",")
-	if len(ips) != 2 {
-		return "", "", errors.Errorf("container addresses should have 2 values, got %d values", len(ips))
+
+	ip := ""
+	for _, net := range containerInfo.NetworkSettings.Networks {
+		ip = net.IPAddress
+		break
 	}
-	return ips[0], ips[1], nil
+	return ip, containerInfo.NetworkSettings.GlobalIPv6Address, nil
```

Comment:
The old inspect command used the `GlobalIPv6Address` in the same array element as `IPAddress`. The new code is using the `GlobalIPv6Address` of the `DefaultNetworkSettings`, is this intended?

Reply:
Oh, good catch. That was my mistake misreading the jq string. Will update!